pkgs/by-name/wg/wgpu-native/package.nix

```diff
@@ -0,0 +1,78 @@
+{
+  rustPlatform,
+  llvmPackages,
+  lib,
+  fetchgit,
+  nix-update-script,
+  cmake,
+  clang,
+  rustc,
+  glfw,
+  meson,
+  pkg-config,
+  fontconfig,
+  stdenv,
+  darwin,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "wgpu-native";
+  version = "22.1.0.5";
+
+  src = fetchgit {
+    url = "https://github.com/gfx-rs/wgpu-native";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-lEUHRU7+sFWtEYTOB2F+SmMNG8nrjro3IL7BgYuIGgM=";
+    fetchSubmodules = true;
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "d3d12-22.0.0" = "sha256-Gtq0xYZoWNwW+BKVLqVVKGqc+4HjaD7NN1hlzyFP5g0=";
+    };
+  };
+
+  nativeBuildInputs = [
+    clang
+    glfw
+    rustc
+  ];
+
+  buildInputs =
+    [ fontconfig ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin (
+      with darwin.apple_sdk.frameworks;
+      [
+        CoreServices
+        QuartzCore
+        AppKit
+      ]
+    );
+
+  LIBCLANG_PATH = "${llvmPackages.libclang.lib}/lib";
+  doCheck = false;
```

Comment:
Why?

1. If there are only parts of tests failing, try disabling them. https://nixos.org/manual/nixpkgs/unstable/#disabling-package-tests
2. If that's not a case, add an annotation here to indicate why you disabled tests.

Reply:
You are right, since it's my first package I used an old pr for this package as a template and just left it there. Right now I'm trying to see if they run succesfully because they might use the GPU